src/server/services/EventService.ts

```diff
@@ -319,6 +319,22 @@ export default class EventService {
     return newEvent.save();
   }
 
+  /**
+   * Edits an existing event with given attributes.
+   * @param request The request with event parameters.
+   * @param logoPath The file path of the logo to associate with the new event.
+   */
+  async editExistingEvent(id: string, request: Partial<RegisterEventRequest>, logoPath?: string) {
+    const event = await this.EventModel.findByIdAndUpdate(id, request);
+    if (logoPath) {
+      try {
+        await event.attach('logo', { path: logoPath });
+      } catch (err) {
+        throw new Error(ErrorMessage.DATABASE_ERROR());
+      }
+    }
+  }
```

Comment:
You are missing an `event.save()` here - probably `return event.save()` because of my previous comment on the API responding with an event.

Reply:
I personally was not able to get the event logo to change either - maybe it is because of the missing `.save()`, not sure.

Reply:
👍 Verified it now saves the image with a `.save()` call on 87d4e2b